Source/WebCore/rendering/RenderObject.h

```diff
@@ -507,6 +507,10 @@ class RenderObject : public CachedImageClient, public CanMakeCheckedPtr {
     // respecting clipping, masking, filters, opacity, stroke-width and markers
     virtual FloatRect repaintRectInLocalCoordinates() const;
 
+    // repaintRectInLocalCoordinates, but more precise, "sensitive to hit detection".
+    // https://svgwg.org/svg2-draft/struct.html#__svg__SVGSVGElement__checkIntersection
+    virtual FloatRect repaintRectInLocalCoordinatesForHitTesting() const;
```

Comment:
The conflation of repaint and hit-testing here is troubling. For example, an SVG element with a drop shadow filter can have a large repaint rect, but the drop-shadow part should not hit-test. So I don't think this is about repaint at all. Can we call it `approximateBoundingBox` or something?

Reply:
Maybe we should just use approximate path bounding box logic inside `repaintRectInLocalCoordinates()`?

Reply:
It'd be great if we didn't pollute generic renderers with this SVG specific behavior and instead keep it within SVG codebase if possible. Also we've got so many repaintRect functions here in RenderObject (public functions), why is it specific to the "inLocalCoordinates" flavor? Is it because that's what SVG ends up calling?

Reply:
`repaintRectInLocalCoordinates` is the approximate repainting rect. And now, `checkInsertion` etc. is using `repaintRectInLocalCoordinates` while it requires "sensitive to hit detection" rect. And this is why we have `repaintRectInLocalCoordinatesForHitTesting` here, which is preserving what we are doing right now.
The code inside `repaintRectInLocalCoordinatesForHitTesting` is not new logic, which is exactly the same to what we are currently doing.